### PR TITLE
`ScaffoldBackgroundColor` should default to `ColorScheme.surface`

### LINKED ISSUE #149158: Changing app background color requires deprecated attribute `background`
### Steps to reproduce

1. Run the example code

### Expected results

Setting the `surface` attribute of the color scheme to `Colors.white` makes the whole app have a white background.

### Actual results

Only the app bar turns white. To make the whole app have a white background, `background` has to be set as well. However, that has been deprecated and the proposed migration path is to use `surface` instead.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(
          seedColor: Colors.deepPurple,
          surface: Colors.white,
        ),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headlineMedium,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}

```

</details>

### Screenshots or Video

_No response_

### Logs

_No response_

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.22.1, on Arch Linux 6.9.1-arch1-2, locale en_US.UTF-8)
    • Flutter version 3.22.1 on channel stable at /home/felipe/snap/flutter/common/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a14f74ff3a (6 days ago), 2024-05-22 11:08:21 -0500
    • Engine revision 55eae6864b
    • Dart version 3.4.1
    • DevTools version 2.34.3

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /home/felipe/Android/Sdk/
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /home/felipe/Applications/android-studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = chromium

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 2022.2)
    • Android Studio at /home/felipe/Applications/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] Connected device (3 available)
    • Pixel 8 (mobile) • 3A101FDJH00722 • android-arm64  • Android 14 (API 34)
    • Linux (desktop)  • linux          • linux-x64      • Arch Linux 6.9.1-arch1-2
    • Chrome (web)     • chrome         • web-javascript • Chromium 125.0.6422.112 Arch Linux

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by QuncCccccc**
`ThemeData.ScaffoldBackgroundColor` originally defaults to `ColorScheme.background`. As background in ColorScheme is deprecated, customizing background is not able to change scaffold background color. This PR is to make `scaffoldBackgroundColor` defaults to `ColorScheme.surface` which is the replacement of `ColorScheme.background`.

Fixes https://github.com/flutter/flutter/issues/149158

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/test/material/dialog_test.dart [hunk 1]
```diff
@@ -111,6 +111,28 @@ void main() {
     expect(materialWidget.color, customColor);
   });
 
+  testWidgets('dialogBackground defaults to ColorScheme.surface', (WidgetTester tester) async {
```
Comment (TahaTesser) [inline packages/flutter/test/material/dialog_test.dart:114]:
```suggestion
  testWidgets('Dialog background defaults to ColorScheme.surface', (WidgetTester tester) async {
```

File: packages/flutter/test/material/scaffold_test.dart [hunk 2]
```diff
@@ -3395,6 +3395,27 @@ void main() {
     // FAB is not visible.
     expect(find.byType(FloatingActionButton), findsNothing);
   });
+
+  testWidgets('scaffoldBackgroundColor defaults to ColorScheme.surface', (WidgetTester tester) async {
```
Comment (TahaTesser) [inline packages/flutter/test/material/scaffold_test.dart:3399]:
```suggestion
  testWidgets('Scaffold background color defaults to ColorScheme.surface', (WidgetTester tester) async {
```

File: packages/flutter/test/material/dialog_test.dart [hunk 3]
```diff
@@ -111,6 +111,28 @@ void main() {
...
+    const Dialog dialog = Dialog(
```
Comment (TahaTesser) [inline packages/flutter/test/material/dialog_test.dart:115]:
```suggestion
    final ThemeData theme = ThemeData(colorScheme: ThemeData().colorScheme.copyWith(
      surface: Colors.orange,
      background: Colors.green,
    ));
    const Dialog dialog = Dialog(
```

Theme variable can be passed to the `MaterialApp` and Dialog background can be tested against `surface` from the `theme` variable.

File: packages/flutter/test/material/dialog_test.dart [hunk 4]
```diff
@@ -111,6 +111,28 @@ void main() {
...
+      child: SizedBox(
+        width: 200,
+        height: 200
+      ),
+    );
+    await tester.pumpWidget(_buildAppWithDialog(
+      dialog,
+      theme: ThemeData(colorScheme: ThemeData().colorScheme.copyWith(
+        surface: Colors.orange,
+      ))
```
Comment (TahaTesser) [inline packages/flutter/test/material/dialog_test.dart:125]:
```suggestion
      theme: theme,
```

File: packages/flutter/test/material/dialog_test.dart [hunk 5]
```diff
@@ -111,6 +111,28 @@ void main() {
...
+    ));
+
+    await tester.tap(find.text('X'));
+    await tester.pumpAndSettle();
+
+    final Material materialWidget = _getMaterialFromDialog(tester);
+    expect(materialWidget.color, Colors.orange);
```
Comment (TahaTesser) [inline packages/flutter/test/material/dialog_test.dart:132]:
```suggestion
    expect(materialWidget.color, theme.colorScheme.surface);
```

File: packages/flutter/test/material/dialog_test.dart [hunk 6]
```diff
@@ -111,6 +111,28 @@ void main() {
...
+  });
+
+
```
Comment (TahaTesser) [inline packages/flutter/test/material/dialog_test.dart:135]:
```suggestion
```

Extra line

File: packages/flutter/test/material/scaffold_test.dart [hunk 7]
```diff
@@ -3395,6 +3395,27 @@ void main() {
...
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(
+          colorScheme: ThemeData().colorScheme.copyWith(
+            surface: Colors.orange,
+            background: Colors.green,
+          )),
```
Comment (TahaTesser) [inline packages/flutter/test/material/scaffold_test.dart:3406]: Similar to `Dialog` test theme.

**Review by TahaTesser — COMMENTED** [review]
Love this change.

**Review by Piinks — COMMENTED** [review]
Was there a migration guide for ColorScheme.background when it was deprecated? If not, we should probably make one, as this change might break folks and they won't know how to readily fix it.

**Comment by QuncCccccc** [comment]
> Was there a migration guide for ColorScheme.background when it was deprecated? If not, we should probably make one, as this change might break folks and they won't know how to readily fix it.

Yes, the migration guide is included in the breaking change page: https://docs.flutter.dev/release/breaking-changes/new-color-scheme-roles :)

**Review by TahaTesser — APPROVED** [review]
LGTM!

**Comment by YazeedAlKhalaf** [comment]
> **NOTE:** Not sure were to put this, but I came from this issue: https://github.com/flutter/flutter/issues/149158

I think a lot of apps using material have a `background` color with a different `surface` color above it. Take for example an app with a scaffold and then on top of it a card, that would break badly.

I found out the description of the `surface` color is the following:
```
/// The background color for widgets like [Card].
final Color surface;
```

I think there is one missing piece, before we had `surface` and `onSurface`. But now there is no replacement, I believe that apps using it are now stuck, unless they create a color extension to get that functionality back.

I think `background` -> `surface` doesn't make sense because they are two different colors.
The same thing goes on `onBackground` -> `onSurface`

> I attached the docs above that are still saying, I copied them from `3.22.0`, `surface` is the background color for cards.

**Comment by QuncCccccc** [comment]
> > **NOTE:** Not sure were to put this, but I came from this issue: #149158
> 
> I think a lot of apps using material have a `background` color with a different `surface` color above it. Take for example an app with a scaffold and then on top of it a card, that would break badly.
> 
> I think there is one missing piece, before we had `surface` and `onSurface`. But now there is no replacement, I believe that apps using it are now stuck, unless they create a color extension to get that functionality back.
> 
> I think `background` -> `surface` doesn't make sense because they are two different colors. The same thing goes on `onBackground` -> `onSurface`
> 
> > I attached the docs above that are still saying, I copied them from `3.22.0`, `surface` is the background color for cards.

Thanks a lot for your suggestions! We are replacing `surface` and `onSurface` with `background` and `onBackground` because the latest Material Design 3 has deprecated these colors. And currently there should be no M3 widgets use `background` and `onBackground` as the defaults. The `surface` and `onSurface` should be able to replace them well if we use the M3 dynamic colors(`ColorScheme.fromSeed` or `ColorScheme.fromImageProvider`) because these two are basically duplicated.

> I found out the description of the `surface` color is the following:
> 
> ```
> /// The background color for widgets like [Card].
> final Color surface;

Thanks for pointing this out! This doc should be updated based on the new `ColorScheme` change because after we added more tone-based surface color roles(like surfaceContainer/-Low/-Lowest/-High/-Highest), `surfaceContainerLow` becomes the default for `Card` and we can see some contrast from the scaffold background. Sorry for the confusion!

**Comment by ishaanbahal** [comment]
Got a surprise on 3.24 like @YazeedAlKhalaf mentioned, we are not using M3 design guidelines strictly, rather using a different background and surface. Anyway, for anyone coming here because of the surprise after update, please set your `scaffoldBackgroundColor` in main ThemeData to revert to your original look:

```dart
final _m3Theme = ThemeData(
    useMaterial3: true,
    scaffoldBackgroundColor: Color(0xFFFAFAFA), // Or a color from scheme you don't generally use, as that helps you keep dart and light modes in check
  );
```

Flutter upgrades generally bring these surprises which require some work to fix, and have to generally go through the changelog every time to see what broke and why, even the migration pages only work if you know how to find them. The doc mentioned above is not part of the breaking-changes landing page [here](https://docs.flutter.dev/release/breaking-changes).

Only reason I upgraded so soon, is I was personally waiting for (PopScope Generic Types)[https://docs.flutter.dev/release/breaking-changes/popscope-with-result] which will resolve https://github.com/flutter/flutter/issues/138737

**Pull request merged**